zarr/hierarchy.py

```diff
@@ -414,6 +416,78 @@ def arrays(self):
                                  chunk_store=self._chunk_store,
                                  synchronizer=self._synchronizer)
 
+    def visit(self, func):
+        """Run callable on each object name.
```

Comment:
Suggest to replace ``callable`` with `` `func` ``. Also maybe worth adding a note about how the return value is handled, e.g., from h5py source: "Returning None continues iteration; returning anything else aborts iteration and returns that value."